src/core/border_router/routing_manager.cpp

```diff
@@ -262,6 +259,8 @@ void RoutingManager::RecvIcmp6Message(uint32_t            aInfraIfIndex,
         HandleRouterAdvertisement(aSrcAddress, aBuffer, aBufferLength);
         break;
     case Ip6::Icmp::Header::kTypeRouterSolicit:
+        // Drop Router Solicitations initiated from infra interface.
+        VerifyOrExit(aSrcAddress != *infraLinkLocalAddr, error = kErrorDrop);
```

Comment:
Does IPv6 allows multiple link local addresses on a single network interface? Is it safer to use network interface index instead of the link local address here?

Reply:
1. I don't think it is possible to get the network interface index of the sender.
2. Even if we can, the sender network interface index can not be used for filtering the packet, because network interface on host A and host B can have the same index.
3. It is possible for an interface to have multiple LLA, but this should be rare. I didn't observe multiple LLAs on a single interface.

Reply:
I meant can we have a function like `FindNetifIndex(aSrcAddress)`, and compare it with InfraLink's index.

Reply:
Thanks, I think it works in this way. May address it in separate PR I think.